stan/math/opencl/prim/cols.hpp

```diff
@@ -16,7 +16,7 @@ namespace math {
  * @return number of columns in x
  */
 template <typename T_x,
-          typename = require_all_kernel_expressions_and_none_scalar_t<T_x>>
+          require_t<is_nonscalar_prim_or_rev_kernel_expression<T_x>>* = nullptr>
```

Comment:
```suggestion
          require_nonscalar_prim_or_rev_kernel_expression<T_x>* = nullptr>
```
Same for rows and size.

Reply:
Done.